src/main/java/reactor/netty/resources/LoopResources.java

```diff
@@ -206,18 +225,18 @@ default EventLoopGroup onClient(boolean useNative) {
 	 * @param group the source {@link EventLoopGroup} to assign a loop from
 	 *
 	 * @return a {@link Class} target for the underlying {@link ServerChannel} factory
+	 * @deprecated as of 0.9.8. See {@link #onChannel(Class, EventLoopGroup)}
 	 */
+	@Deprecated
 	default Class<? extends ServerChannel> onServerChannel(EventLoopGroup group) {
-		return preferNative() ? DefaultLoopNativeDetector.getInstance().getServerChannel(group) :
+		return preferNative() ? onChannel(ServerSocketChannel.class, group).getClass() :
 				NioServerSocketChannel.class;
 	}
 
 	/**
 	 * Create a server select {@link EventLoopGroup} for servers to be used
```

Comment:
There is a difference

`onServerSelect` - Create a **server select** {@link EventLoopGroup}, which means the event loop group for the server acceptor channel, in Netty this is the `boss` group.

`onServer` - Callback for **server** {@link EventLoopGroup}, which means the event loop group for the child channel that will process the request, in Netty this is the `worker` group.

see here
https://github.com/netty/netty/blob/387e451c8289f05bf553821170bde30052101f8a/transport/src/main/java/io/netty/bootstrap/ServerBootstrap.java#L218
https://github.com/netty/netty/blob/387e451c8289f05bf553821170bde30052101f8a/transport/src/main/java/io/netty/bootstrap/ServerBootstrap.java#L154

In `LoopResources`, `onClient` and `onServerSelect` are aliases for `onServer`, but in the actual implementation - 
`DefaultLoopResources`, the implementation is different depending on the configuration i.e. number of the worker and selector threads (that's for the server), while for the client we create `ColocatedEventLoopGroup` so that when in one JVM the server and the client will share the workers.

Reply:
I extended the javadoc